### PR TITLE
MAINT: Chaining exception in numpy\numpy\ma\mrecords.py

### PR DESCRIPTION
**Pull request by Ashutosh619-sudo**
Changes related to #15986

added chaining for Attribute error in mrecords.py.

### CODE REVIEW

File: numpy/linalg/lapack_lite/make_lite.py [hunk 1]
```diff
@@ -262,7 +262,7 @@ def runF2C(fortran_filename, output_dir):
             ["f2c"] + F2C_ARGS + ['-d', output_dir, fortran_filename]
         )
     except subprocess.CalledProcessError:
-        raise F2CError
+        raise F2CError from None
```
Comment (eric-wieser) [inline numpy/linalg/lapack_lite/make_lite.py:265]: This one would probably be better as `from e`

File: numpy/ma/mrecords.py [hunk 2]
```diff
@@ -198,8 +198,8 @@ def __getattribute__(self, attr):
         fielddict = ndarray.__getattribute__(self, 'dtype').fields
         try:
             res = fielddict[attr][:2]
-        except (TypeError, KeyError):
-            raise AttributeError("record array has no attribute %s" % attr)
+        except (TypeError, KeyError) as e:
+            raise AttributeError("record array has no attribute %s" % attr) from e
```
Comment (eric-wieser) [inline numpy/ma/mrecords.py:202]: This matches the following, so is fine, thanks!

https://github.com/numpy/numpy/blob/432b3899dea324d82e3b7aa79355934442f97cec/numpy/core/records.py#L464-L465

File: numpy/linalg/lapack_lite/make_lite.py (hunk 1 above)
Comment (Ashutosh619-sudo) [reply numpy/linalg/lapack_lite/make_lite.py:265]: Okk I'll do it.

Comment (mattip) [reply numpy/linalg/lapack_lite/make_lite.py]: What happened here? Was this removed on purpose?

Comment (Ashutosh619-sudo) [reply numpy/linalg/lapack_lite/make_lite.py]: Yes it was removed on purpose, I will correct it in another PR.

**Comment by mattip** [comment]
Thanks @Ashutosh619-sudo

**Comment by Ashutosh619-sudo** [comment]
@mattip can you please add the label hacktoberfest-accepted as I have participated in the hackertoberfest and unless you do this it will not count as an PR count.

**Comment by eric-wieser** [comment]
@Ashutosh619-sudo: The hacktoberfest website says that doing that is equivalent to merging, so there should be no need

**Comment by eric-wieser** [comment]
Ah, I take that back - I now see it says:

> Alternatively, an individual PR can be opted-in with a maintainer adding the "hacktoberfest-accepted" label to the PR.

**Comment by Ashutosh619-sudo** [comment]
No they given a new update that only repo with topic hacktoberfest or PR labelled as hacktoberfest-accepted will be counted, to reduce spam.

**Comment by eric-wieser** [comment]
Yep, it changed since I last looked

@mattip, do you have any preference between creating the topic and creating the label? The former might draw in more noise PRs, while the latter leaves a label dangling in our issue tracker.

**Comment by Ashutosh619-sudo** [comment]
I would suggest to creating the label and not labelling every PR with it but labelling the one's in which the contributor ask for it.

**Comment by mattip** [comment]
We don't have a stated policy on hacktoberfest. Maybe we should discuss it at the upcoming community meeting. I don't really feel like encouraging Digital Ocean's misguided offer to "help" OSS projects by swamping reviewers and maintainers with once-off contributions.

**Pull request merged**